File: TI1/TI1.4 RESTFUL_DOTNET_SINBDD_GR09/02.CLIESC/CONUNI_RESTFUL_DOTNET_CLIESC_G09/CONUNI_RESTFUL_DOTNET_CLIESC_G09/vista/PanelResultado.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace CONUNI_RESTFUL_DOTNET_CLIESC_G09.vista
{
    /// <summary>
    /// Panel para mostrar resultados de conversiones - Diseño compacto y funcional
    /// </summary>
    public class PanelResultado : Panel
    {
        private static readonly Color AZUL_PRINCIPAL = Color.FromArgb(33, 169, 218);
        private static readonly Color ROJO_MONSTER = Color.FromArgb(246, 105, 113);
        private static readonly Color AMARILLO_MONSTER = Color.FromArgb(245, 216, 128);
        private static readonly Color PURPURA_MONSTER = Color.FromArgb(140, 107, 205);
        private static readonly Color AZUL_CLARO = Color.FromArgb(159, 220, 250);
        private static readonly Color BLANCO = Color.White;
        private static readonly Color GRIS_OSCURO = Color.FromArgb(60, 60, 60);
        private static readonly Color GRIS_CLARO = Color.FromArgb(240, 240, 240);

        private Label lblTitulo;
        private TextBox areaResultado;

        public PanelResultado()
        {
            InicializarComponentes();
            ConfigurarLayout();
        }

        private void InicializarComponentes()
        {
            lblTitulo = new Label();
            lblTitulo.Text = "RESULTADO DE LA CONVERSIÓN";
            lblTitulo.Font = new Font("Arial", 16F, FontStyle.Bold);
            lblTitulo.ForeColor = AZUL_PRINCIPAL;
            lblTitulo.TextAlign = ContentAlignment.MiddleCenter;
            lblTitulo.Dock = DockStyle.Top;

            areaResultado = new TextBox();
            areaResultado.Font = new Font("Arial", 14F);
            areaResultado.ForeColor = GRIS_OSCURO;
            areaResultado.BackColor = Color.FromArgb(248, 249, 250);
            areaResultado.Multiline = true;
            areaResultado.ReadOnly = true;
            areaResultado.ScrollBars = ScrollBars.Vertical;
            areaResultado.WordWrap = true;
            areaResultado.Text = "El resultado de la conversión aparecerá aquí...";
            areaResultado.Dock = DockStyle.Fill;
            areaResultado.BorderStyle = BorderStyle.None;
            areaResultado.Padding = new Padding(20);
        }

        private void ConfigurarLayout()
        {
            this.BackColor = BLANCO;
            this.Dock = DockStyle.Fill;
            this.Padding = new Padding(10);

            this.Controls.Add(areaResultado);
            this.Controls.Add(lblTitulo);
        }

        public void MostrarResultado(string resultado)
        {
            areaResultado.Text = resultado;
            areaResultado.ForeColor = AZUL_PRINCIPAL;
            areaResultado.BackColor = AZUL_CLARO;
            areaResultado.Font = new Font("Arial", 16F, FontStyle.Bold);
        }

        public void MostrarError(string error)
        {
            areaResultado.Text = "ERROR: " + error;
            areaResultado.ForeColor = ROJO_MONSTER;
            areaResultado.BackColor = Color.FromArgb(255, 240, 240);
            areaResultado.Font = new Font("Arial", 14F, FontStyle.Bold);
        }

        public void Limpiar()
        {
            areaResultado.Text = "El resultado de la conversión aparecerá aquí...";
            areaResultado.ForeColor = GRIS_OSCURO;
            areaResultado.BackColor = Color.FromArgb(248, 249, 250);
            areaResultado.Font = new Font("Arial", 14F);
        }
    }
}

File: Proyectos/TI.1 PAO202650_SINBDD_NRC30742/TI.1.3. PAO202650_RESTFUL_JAVA_SINBDD_NRC30742/04. CLIESC/CONUNI_SOAP_JAVA_CLIESC_G05/src/main/java/ec/edu/monster/vista/PanelResultado.java
package ec.edu.monster.vista;

import javax.swing.*;
import java.awt.*;

/**
 * Panel para mostrar resultados de conversiones - Diseño compacto y funcional
 * @author ACER NITRO V15
 */
public class PanelResultado extends JPanel {
    
    // Colores oficiales de Monsters Inc.
    private static final Color AZUL_PRINCIPAL = new Color(33, 169, 218);      // #21a9da
    private static final Color ROJO_MONSTER = new Color(246, 105, 113);        // #f66971
    private static final Color AMARILLO_MONSTER = new Color(245, 216, 128);   // #f5d880
    private static final Color PURPURA_MONSTER = new Color(140, 107, 205);     // #8c6bcd
    private static final Color AZUL_CLARO = new Color(159, 220, 250);         // #9fdcfa
    private static final Color BLANCO = new Color(255, 255, 255);
    private static final Color GRIS_OSCURO = new Color(60, 60, 60);
    private static final Color GRIS_CLARO = new Color(240, 240, 240);
    
    // Componentes
    private JLabel lblTitulo;
    private JTextArea areaResultado;
    
    public PanelResultado() {
        inicializarComponentes();
        configurarLayout();
    }
    
    private void inicializarComponentes() {
        // Título
        lblTitulo = new JLabel("RESULTADO DE LA CONVERSIÓN");
        lblTitulo.setFont(new Font("Arial", Font.BOLD, 16));
        lblTitulo.setForeground(AZUL_PRINCIPAL);
        lblTitulo.setHorizontalAlignment(SwingConstants.CENTER);
        
        // Área de resultado
        areaResultado = new JTextArea();
        areaResultado.setFont(new Font("Arial", Font.PLAIN, 14));
        areaResultado.setForeground(GRIS_OSCURO);
        areaResultado.setBackground(new Color(248, 249, 250));
        areaResultado.setBorder(BorderFactory.createEmptyBorder(20, 20, 20, 20));
        areaResultado.setEditable(false);
        areaResultado.setWrapStyleWord(true);
        areaResultado.setLineWrap(true);
        areaResultado.setText("El resultado de la conversión aparecerá aquí...");
        areaResultado.setAlignmentX(Component.CENTER_ALIGNMENT);
    }
    
    private void configurarLayout() {
        setBackground(BLANCO);
        setLayout(new BorderLayout());
        
        // Título
        add(lblTitulo, BorderLayout.NORTH);
        
        // Área de resultado
        add(areaResultado, BorderLayout.CENTER);
    }
    
    public void mostrarResultado(String resultado) {
        System.out.println("PanelResultado - Recibido resultado: " + resultado);
        areaResultado.setText(resultado);
        areaResultado.setForeground(AZUL_PRINCIPAL);
        areaResultado.setBackground(AZUL_CLARO);
        areaResultado.setFont(new Font("Arial", Font.BOLD, 16));
        areaResultado.setBorder(BorderFactory.createCompoundBorder(
            BorderFactory.createLineBorder(AZUL_PRINCIPAL, 2),
            BorderFactory.createEmptyBorder(20, 20, 20, 20)
        ));
        System.out.println("PanelResultado - Resultado mostrado correctamente");
    }
    
    public void mostrarError(String error) {
        System.out.println("PanelResultado - Recibido error: " + error);
        areaResultado.setText("ERROR: " + error);
        areaResultado.setForeground(ROJO_MONSTER);
        areaResultado.setBackground(new Color(255, 240, 240));
        areaResultado.setFont(new Font("Arial", Font.BOLD, 14));
        areaResultado.setBorder(BorderFactory.createCompoundBorder(
            BorderFactory.createLineBorder(Color.RED, 2),
            BorderFactory.createEmptyBorder(20, 20, 20, 20)
        ));
        System.out.println("PanelResultado - Error mostrado correctamente");
    }
    
    public void limpiar() {
        areaResultado.setText("El resultado de la conversión aparecerá aquí...");
        areaResultado.setForeground(GRIS_OSCURO);
        areaResultado.setBackground(new Color(248, 249, 250));
        areaResultado.setFont(new Font("Arial", Font.PLAIN, 14));
        areaResultado.setBorder(BorderFactory.createEmptyBorder(20, 20, 20, 20));
    }
}